src/AaveCurator/AaveCurator.sol

```diff
@@ -0,0 +1,301 @@
+    Standard,
+    BPT8020
+  }
+
+  struct TokenChailinkOracleData {
+    address oracle;
+    bool isEthBased;
+  }
+
+  modifier onlyOwnerOrManager() {
+    if (msg.sender != owner() && msg.sender != manager) {
+      revert InvalidCaller();
+    }
+    _;
+  }
+
+  IRocketPoolDeposit private constant ROCKET_POOL =
+    IRocketPoolDeposit(0xDD3f50F8A6CafbE9b31a427582963f465E745AF8);
+  IWeth private constant WETH = IWeth(AaveV2EthereumAssets.WETH_UNDERLYING);
+
+  address public constant STETH = AaveV2EthereumAssets.stETH_UNDERLYING;
+  address public constant BPT_PRICE_CHECKER = 0x7961bBC81352F26d073aA795EED51290C350D404; // TODO: Update with new one
+  address public constant CHAINLINK_PRICE_CHECKER = 0x4D2c3773E69cB69963bFd376e538eC754409ACFa;
+
+  address public milkman = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public manager;
+
+  mapping(address tokenAddress => bool) public allowedFromTokens;
+  mapping(address tokenAddress => bool) public allowedToTokens;
+  mapping(address tokenAddress => TokenChailinkOracleData) public tokenChainlinkOracle;
+
+  constructor() Ownable() {}
+
+  function initialize() external initializer {}
+
+  function swap(
+    address fromToken,
+    address toToken,
+    address recipient,
+    uint256 amount,
+    uint256 slippage,
+    TokenType _tokenType
+  ) external onlyOwnerOrManager {
+    if (amount == 0) revert InvalidAmount();
+    if (!allowedFromTokens[fromToken]) revert InvalidToken();
+    if (!allowedToTokens[toToken]) revert InvalidToken();
+    if (recipient != address(this) && recipient != address(AaveV3Ethereum.COLLECTOR)) {
+      revert InvalidRecipient();
+    }
+
+    IERC20(fromToken).safeApprove(milkman, amount);
+
+    (address priceChecker, bytes memory data) = _getPriceCheckerAndData(
+      _tokenType,
+      slippage,
+      fromToken,
+      toToken
+    );
+
+    IMilkman(milkman).requestSwapExactTokensForTokens(
+      amount,
+      IERC20(fromToken),
+      IERC20(toToken),
+      recipient,
+      priceChecker,
+      data
+    );
+
+    emit SwapRequested(fromToken, toToken, amount);
+  }
+
+  function cancelSwap(
+    address tradeMilkman,
+    address fromToken,
+    address toToken,
+    address recipient,
+    uint256 amount,
+    uint256 slippage,
+    TokenType _tokenType
+  ) external onlyOwnerOrManager {
+    (address priceChecker, bytes memory data) = _getPriceCheckerAndData(
+      _tokenType,
+      slippage,
+      fromToken,
+      toToken
+    );
+
+    IMilkman(tradeMilkman).cancelSwap(
+      amount,
+      IERC20(fromToken),
+      IERC20(toToken),
+      recipient,
+      priceChecker,
+      data
+    );
+
+    IERC20(fromToken).safeTransfer(
+      address(AaveV3Ethereum.COLLECTOR),
+      IERC20(fromToken).balanceOf(address(this))
+    );
+
+    emit SwapCanceled(fromToken, toToken, amount);
+  }
+
+  /// @notice Withdraws wETH into ETH, then deposits for wstETH
+  function depositWstETH(uint256 amount) external onlyOwnerOrManager {
```

Comment:
how much sense does this make (same with rETH)

Wouldn't it be more reasonable to just put this into a library (e.g. on aave helpers)? What's the benefit of having it on an additional external contract?

Reply:
Probably a good idea to have it on aave-helpers instead.